src/audio/tdfb/tdfb_direction.c

```diff
@@ -0,0 +1,497 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+//#define DEBUG
+
+#include <ipc/topology.h>
+#include <sof/audio/eq_iir/iir.h>
+#include <sof/audio/tdfb/tdfb_comp.h>
+#include <sof/lib/alloc.h>
+#include <sof/math/iir_df2t.h>
+#include <sof/math/trig.h>
+#include <user/eq.h>
+#include <stdint.h>
+#include "tdfb_math.h"
+
+#ifdef DEBUG
+#include <stdio.h>
+FILE *fh;
+#endif
+
+/* Generic definitions */
+#define ZERO_POINT_FIVE		Q_CONVERT_FLOAT(0.5, 30)
+#define SPEED_OF_SOUND		340				/* m/s Q0 */
```

Comment:
Should this be in math.h as others may need it ? Also assume at 1000mb ?

Reply:
Yes, seems that is should be 343 m/s at one bar and 20C!